File: week01/DAYE/1859.java
import java.util.Scanner;
import java.io.FileInputStream;
 
class Solution
{
    public static void main(String args[]) throws Exception
    {
        Scanner sc = new Scanner(System.in);
        int T;
        T=sc.nextInt();
 
        for(int test_case = 1; test_case <= T; test_case++){
            int N = sc.nextInt();
            int[] price = new int[N];
 
            for (int i = 0; i < N; i++) {
                price[i] = sc.nextInt();
            }
            long answer = 0;    
            int maxPrice = 0;
            for (int i = N - 1; i >= 0; i--) {
                if (price[i] > maxPrice) {
                    maxPrice = price[i];
                } else {
                    answer += maxPrice - price[i];
                }
            }
            System.out.println("#" + test_case + " " + answer);
        }
    }
}

File: week01/DAYE/1859.py
T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    N = int(input())
    price = list(map(int, input().split()))
 
    answer = 0
    max_price = 0
 
    for i in range(N - 1, -1, -1):
        if price[i] > max_price:
            max_price = price[i]
        else:
            answer += max_price - price[i]
 
    print(f"#{test_case} {answer}")